error in compiling mrpt master ; "cmake --build ."

hello,
ubuntu 21.10 aarch64 on raspberry_pi 4B 8Go

```
/home/pi/mrpt/python/src/maps_bindings.cpp: In function ‘mrpt::opengl::CSetOfObjects::Ptr CMetricMap_getAs3DObject(mrpt::maps::CMetricMap&)’:
/home/pi/mrpt/python/src/maps_bindings.cpp:53:14: error: ‘class mrpt::maps::CMetricMap’ has no member named ‘getAs3DObject’
   53 |         self.getAs3DObject(outObj);
      |              ^~~~~~~~~~~~~
/home/pi/mrpt/python/src/maps_bindings.cpp: In function ‘mrpt::opengl::CSetOfObjects::Ptr CPointsMap_getAs3DObject(mrpt::maps::CPointsMap&)’:
/home/pi/mrpt/python/src/maps_bindings.cpp:183:14: error: ‘class mrpt::maps::CPointsMap’ has no member named ‘getAs3DObject’
  183 |         self.getAs3DObject(outObj);
      |              ^~~~~~~~~~~~~
/home/pi/mrpt/python/src/maps_bindings.cpp: In function ‘mrpt::opengl::CSetOfObjects::Ptr CMultiMetricMap_getAs3DObject(mrpt::maps::CMultiMetricMap&)’:
/home/pi/mrpt/python/src/maps_bindings.cpp:287:14: error: ‘class mrpt::maps::CMultiMetricMap’ has no member named ‘getAs3DObject’
  287 |         self.getAs3DObject(outObj);
      |              ^~~~~~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-unused-local-typedef’ may have been intended to silence earlier diagnostics
gmake[2]: *** [python/CMakeFiles/pymrpt.dir/build.make:147 : python/CMakeFiles/pymrpt.dir/src/maps_bindings.cpp.o] Erreur 1
gmake[1]: *** [CMakeFiles/Makefile2:12185 : python/CMakeFiles/pymrpt.dir/all] Erreur 2
gmake: *** [Makefile:182 : all] Erreur 2
```

--- a/cmakemodules/script_python_bindings.cmake
+++ b/cmakemodules/script_python_bindings.cmake
@@ -13,7 +13,7 @@ endif()
 if(UNIX AND NOT MRPT_DISABLE_PYTHON_BINDINGS)
     #set( BUILD_PY_BINDINGS OFF CACHE BOOL "If you want to build the MRPT python bindings, enable this.")
     # find packages quiet
-    find_package(Boost QUIET COMPONENTS python3)
+    find_package(Boost QUIET COMPONENTS python)
 
 	# Requires CMake 3.13+
 	find_package (Python3 COMPONENTS Development QUIET)

--- a/python/include/bindings.h
+++ b/python/include/bindings.h
@@ -55,35 +55,26 @@
 		vector_indexing_suite<std::vector<class_name>>());
 
 #define MAKE_PTR_CTX(class_name)                                               \
-	class_name& class_name##Ptr_get_ctx(class_name::Ptr& self)                 \
+	class_name& class_name##Ptr_get_ctx(class_name::Ptr& me) { return *me; }   \
+	void class_name##Ptr_set_ctx(class_name::Ptr& me, class_name& ctx)         \
 	{                                                                          \
-		return *self;                                                          \
+		*me = ctx;                                                             \
 	}                                                                          \
-	void class_name##Ptr_set_ctx(class_name::Ptr& self, class_name& ctx)       \
+	class_name* class_name##Ptr_pointer(class_name::Ptr& me)                   \
 	{                                                                          \
-		*self = ctx;                                                           \
-	}                                                                          \
-	class_name* class_name##Ptr_pointer(class_name::Ptr& self)                 \
-	{                                                                          \
-		return self.get();                                                     \
+		return me.get();                                                       \
 	}
 
 #define MAKE_AS_STR(class_name)                                                \
-	std::string class_name##_asString(class_name& self)                        \
-	{                                                                          \
-		return self.asString();                                                \
-	}
+	std::string class_name##_asString(class_name& me) { return me.asString(); }
 
 #define MAKE_GETITEM(class_name, value_type)                                   \
-	value_type class_name##_getitem(class_name& self, size_t i)                \
-	{                                                                          \
-		return self[i];                                                        \
-	}
+	value_type class_name##_getitem(class_name& me, size_t i) { return me[i]; }
 
 #define MAKE_SETITEM(class_name, value_type)                                   \
-	void class_name##_setitem(class_name& self, size_t i, value_type value)    \
+	void class_name##_setitem(class_name& me, size_t i, value_type value)      \
 	{                                                                          \
-		self[i] = value;                                                       \
+		me[i] = value;                                                         \
 	}
 
 #define MAKE_SUBMODULE(mod)                                                    \

--- a/python/src/bayes_bindings.cpp
+++ b/python/src/bayes_bindings.cpp
@@ -44,12 +44,12 @@ typedef std::deque<CProbabilityParticle<TPose3D, particle_storage_mode::VALUE>>
 
 // CParticleFilter
 CParticleFilter::TParticleFilterStats CParticleFilter_executeOn(
-	CParticleFilter& self, CParticleFilterCapable& obj,
+	CParticleFilter& me, CParticleFilterCapable& obj,
 	const mrpt::obs::CActionCollection::Ptr action,
 	const mrpt::obs::CSensoryFrame::Ptr observation)
 {
 	CParticleFilter::TParticleFilterStats stats;
-	self.executeOn(obj, action.get(), observation.get(), &stats);
+	me.executeOn(obj, action.get(), observation.get(), &stats);
 	return stats;
 }
 // end of CParticleFilter

--- a/python/src/gui_bindings.cpp
+++ b/python/src/gui_bindings.cpp
@@ -19,21 +19,21 @@ using namespace mrpt::opengl;
 
 // CDisplayWindow3D
 tuple CDisplayWindow3D_waitForKey(
-	CDisplayWindow3D& self, bool ignoreControlKeys = true)
+	CDisplayWindow3D& me, bool ignoreControlKeys = true)
 {
 	list ret_val;
 	mrptKeyModifier out_pushModifier = mrptKeyModifier::MRPTKMOD_NONE;
-	int key = self.waitForKey(ignoreControlKeys, &out_pushModifier);
+	int key = me.waitForKey(ignoreControlKeys, &out_pushModifier);
 	ret_val.append(key);
 	ret_val.append(out_pushModifier);
 	return tuple(ret_val);
 }
 
-tuple CDisplayWindow3D_getPushedKey(CDisplayWindow3D& self)
+tuple CDisplayWindow3D_getPushedKey(CDisplayWindow3D& me)
 {
 	list ret_val;
 	mrptKeyModifier out_pushModifier = mrptKeyModifier::MRPTKMOD_NONE;
-	int key = self.getPushedKey(&out_pushModifier);
+	int key = me.getPushedKey(&out_pushModifier);
 	ret_val.append(key);
 	ret_val.append(out_pushModifier);
 	return tuple(ret_val);

--- a/python/src/kinematics_bindings.cpp
+++ b/python/src/kinematics_bindings.cpp
@@ -20,25 +20,24 @@ using namespace mrpt::poses;
 using namespace mrpt::math;
 
 // CVehicleSimul_DiffDriven
-CPose2D CVehicleSimul_DiffDriven_getOdometry(CVehicleSimul_DiffDriven& self)
+CPose2D CVehicleSimul_DiffDriven_getOdometry(CVehicleSimul_DiffDriven& me)
 {
-	return CPose2D(self.getCurrentOdometricPose());
+	return CPose2D(me.getCurrentOdometricPose());
 }
 
-CPose2D CVehicleSimul_DiffDriven_getRealPose(CVehicleSimul_DiffDriven& self)
+CPose2D CVehicleSimul_DiffDriven_getRealPose(CVehicleSimul_DiffDriven& me)
 {
-	return CPose2D(self.getCurrentGTPose());
+	return CPose2D(me.getCurrentGTPose());
 }
 
-TTwist2D CVehicleSimul_DiffDriven_getCurrentGTVel(
-	CVehicleSimul_DiffDriven& self)
+TTwist2D CVehicleSimul_DiffDriven_getCurrentGTVel(CVehicleSimul_DiffDriven& me)
 {
-	return self.getCurrentGTVel();
+	return me.getCurrentGTVel();
 }
 TTwist2D CVehicleSimul_DiffDriven_getCurrentGTVelLocal(
-	CVehicleSimul_DiffDriven& self)
+	CVehicleSimul_DiffDriven& me)
 {
-	return self.getCurrentGTVelLocal();
+	return me.getCurrentGTVelLocal();
 }
 
 BOOST_PYTHON_MEMBER_FUNCTION_OVERLOADS(

--- a/python/src/maps_bindings.cpp
+++ b/python/src/maps_bindings.cpp
@@ -34,24 +34,22 @@ using namespace mrpt::obs;
 
 // CMetricMap
 bool CMetricMap_insertObservation(
-	CMetricMap& self, const CObservation& obs,
+	CMetricMap& me, const CObservation& obs,
 	const CPose3D& robotPose = CPose3D())
 {
-	return self.insertObservation(obs, robotPose);
+	return me.insertObservation(obs, robotPose);
 }
 
 bool CMetricMap_insertObservationPtr(
-	CMetricMap& self, const CObservation::Ptr& obs,
+	CMetricMap& me, const CObservation::Ptr& obs,
 	const CPose3D& robotPose = CPose3D())
 {
-	return self.insertObservationPtr(obs, robotPose);
+	return me.insertObservationPtr(obs, robotPose);
 }
 
-CSetOfObjects::Ptr CMetricMap_getAs3DObject(CMetricMap& self)
+CSetOfObjects::Ptr CMetricMap_getAs3DObject(CMetricMap& me)
 {
-	CSetOfObjects::Ptr outObj = std::make_shared<CSetOfObjects>();
-	self.getAs3DObject(outObj);
-	return outObj;
+	return me.getVisualization();
 }
 BOOST_PYTHON_FUNCTION_OVERLOADS(
 	CMetricMap_insertObservation_overloads, CMetricMap_insertObservation, 2, 3)
@@ -61,19 +59,19 @@ BOOST_PYTHON_FUNCTION_OVERLOADS(
 // end of CMetricMap
 
 // COccupancyGridMap2D
-COccupancyGridMap2D* COccupancyGridMap2D_copy(COccupancyGridMap2D& self)
+COccupancyGridMap2D* COccupancyGridMap2D_copy(COccupancyGridMap2D& me)
 {
-	return (COccupancyGridMap2D*)self.clone();
+	return (COccupancyGridMap2D*)me.clone();
 }
 
 bool COccupancyGridMap2D_insertObservation(
-	COccupancyGridMap2D& self, CObservation& obs, CPose3D& pose)
+	COccupancyGridMap2D& me, CObservation& obs, CPose3D& pose)
 {
-	return self.insertObservation(obs, pose);
+	return me.insertObservation(obs, pose);
 }
 
 object COccupancyGridMap2D_to_ROS_OccupancyGrid_msg1(
-	COccupancyGridMap2D& self, str frame_id)
+	COccupancyGridMap2D& me, str frame_id)
 {
 	// import msg
 	dict locals;
@@ -83,25 +81,25 @@ object COccupancyGridMap2D_to_ROS_OccupancyGrid_msg1(
 		object(), locals);
 	object occupancy_grid_msg = locals["occupancy_grid_msg"];
 	// set info
-	int32_t width = self.getSizeX();
-	int32_t height = self.getSizeY();
+	int32_t width = me.getSizeX();
+	int32_t height = me.getSizeY();
 	occupancy_grid_msg.attr("header").attr("frame_id") = frame_id;
 	occupancy_grid_msg.attr("header").attr("stamp") =
 		TTimeStamp_to_ROS_Time(long_(mrpt::system::now()));
 	occupancy_grid_msg.attr("info").attr("width") = width;
 	occupancy_grid_msg.attr("info").attr("height") = height;
-	occupancy_grid_msg.attr("info").attr("resolution") = self.getResolution();
+	occupancy_grid_msg.attr("info").attr("resolution") = me.getResolution();
 	occupancy_grid_msg.attr("info").attr("origin").attr("position").attr("x") =
-		self.getXMin();
+		me.getXMin();
 	occupancy_grid_msg.attr("info").attr("origin").attr("position").attr("y") =
-		self.getYMin();
+		me.getYMin();
 	// set data
 	boost::python::list data;
 	for (int32_t y = 0; y < height; ++y)
 	{
 		for (int32_t x = 0; x < width; ++x)
 		{
-			float occupancy = self.getCell(x, y);
+			float occupancy = me.getCell(x, y);
 			if (occupancy < 0.45) { data.append(100); }
 			else if (occupancy > 0.55)
 			{
@@ -118,13 +116,13 @@ object COccupancyGridMap2D_to_ROS_OccupancyGrid_msg1(
 }
 
 #ifdef ROS_EXTENSIONS
-object COccupancyGridMap2D_to_ROS_OccupancyGrid_msg2(COccupancyGridMap2D& self)
+object COccupancyGridMap2D_to_ROS_OccupancyGrid_msg2(COccupancyGridMap2D& me)
 {
-	return COccupancyGridMap2D_to_ROS_OccupancyGrid_msg1(self, str("map"));
+	return COccupancyGridMap2D_to_ROS_OccupancyGrid_msg1(me, str("map"));
 }
 
 void COccupancyGridMap2D_from_ROS_OccupancyGrid_msg(
-	COccupancyGridMap2D& self, object occupancy_grid_msg)
+	COccupancyGridMap2D& me, object occupancy_grid_msg)
 {
 	// set info
 	float x_min = extract<float>(occupancy_grid_msg.attr("info")
@@ -143,7 +141,7 @@ void COccupancyGridMap2D_from_ROS_OccupancyGrid_msg(
 		extract<int32_t>(occupancy_grid_msg.attr("info").attr("height"));
 	float x_max = x_min + width * resolution;
 	float y_max = y_min + height * resolution;
-	self.setSize(x_min, x_max, y_min, y_max, resolution);
+	me.setSize(x_min, x_max, y_min, y_max, resolution);
 	// set data
 	int32_t idx = 0;
 	boost::python::list data =
@@ -153,11 +151,10 @@ void COccupancyGridMap2D_from_ROS_OccupancyGrid_msg(
 		for (int32_t x = 0; x < width; ++x)
 		{
 			int32_t occupancy = extract<int32_t>(data[idx]);
-			if (occupancy >= 0)
-			{ self.setCell(x, y, (100 - occupancy) / 100.0); }
+			if (occupancy >= 0) { me.setCell(x, y, (100 - occupancy) / 100.0); }
 			else
 			{
-				self.setCell(x, y, 0.5);
+				me.setCell(x, y, 0.5);
 			}
 			idx++;
 		}
@@ -176,35 +173,32 @@ BOOST_PYTHON_MEMBER_FUNCTION_OVERLOADS(
 // end of COccupancyGridMap2D
 
 // CPointsMap
-mrpt::opengl::CSetOfObjects::Ptr CPointsMap_getAs3DObject(CPointsMap& self)
+mrpt::opengl::CSetOfObjects::Ptr CPointsMap_getAs3DObject(CPointsMap& me)
 {
-	mrpt::opengl::CSetOfObjects::Ptr outObj =
-		mrpt::opengl::CSetOfObjects::Create();
-	self.getAs3DObject(outObj);
-	return outObj;
+	return me.getVisualization();
 }
 // end of CPointsMap
 
 // CSimplePointsMap
 void CSimplePointsMap_loadFromRangeScan1(
-	CSimplePointsMap& self, const CObservation2DRangeScan& rangeScan)
+	CSimplePointsMap& me, const CObservation2DRangeScan& rangeScan)
 {
-	self.loadFromRangeScan(rangeScan);
+	me.loadFromRangeScan(rangeScan);
 }
 
 void CSimplePointsMap_loadFromRangeScan2(
-	CSimplePointsMap& self, const CObservation2DRangeScan& rangeScan,
+	CSimplePointsMap& me, const CObservation2DRangeScan& rangeScan,
 	const CPose3D& robotPose)
 {
-	self.loadFromRangeScan(rangeScan, robotPose);
+	me.loadFromRangeScan(rangeScan, robotPose);
 }
 
 boost::python::tuple CSimplePointsMap_getPointAllFieldsFast(
-	CSimplePointsMap& self, uint32_t index)
+	CSimplePointsMap& me, uint32_t index)
 {
 	boost::python::list ret_val;
 	std::vector<float> point_xyz;
-	self.getPointAllFieldsFast(index, point_xyz);
+	me.getPointAllFieldsFast(index, point_xyz);
 	ret_val.append(point_xyz[0]);
 	ret_val.append(point_xyz[1]);
 	ret_val.append(point_xyz[2]);
@@ -212,26 +206,26 @@ boost::python::tuple CSimplePointsMap_getPointAllFieldsFast(
 }
 
 boost::python::tuple CSimplePointsMap_getPointFast(
-	CSimplePointsMap& self, uint32_t index)
+	CSimplePointsMap& me, uint32_t index)
 {
 	boost::python::list ret_val;
 	float x, y, z;
-	self.getPointFast(index, x, y, z);
+	me.getPointFast(index, x, y, z);
 	ret_val.append(x);
 	ret_val.append(y);
 	ret_val.append(z);
 	return boost::python::tuple(ret_val);
 }
 
-uint32_t CSimplePointsMap_getSize(CSimplePointsMap& self)
+uint32_t CSimplePointsMap_getSize(CSimplePointsMap& me)
 {
-	return self.getPointsBufferRef_x().size();
+	return me.getPointsBufferRef_x().size();
 }
 // end of CSimplePointsMap
 
 // CSimpleMap
 void CSimpleMap_insert(
-	CSimpleMap& self, CPose3DPDF& in_posePDF, CSensoryFrame& in_SF)
+	CSimpleMap& me, CPose3DPDF& in_posePDF, CSensoryFrame& in_SF)
 {
 	// create smart pointers
 	CPose3DPDF::Ptr in_posePDFPtr = std::dynamic_pointer_cast<CPose3DPDF>(
@@ -239,66 +233,62 @@ void CSimpleMap_insert(
 	CSensoryFrame::Ptr in_SFPtr =
 		std::dynamic_pointer_cast<CSensoryFrame>(in_SF.duplicateGetSmartPtr());
 	// insert smart pointers
-	self.insert(in_posePDFPtr, in_SFPtr);
+	me.insert(in_posePDFPtr, in_SFPtr);
 }
 // end of CSimpleMap
 
 // CMultiMetricMapPDF
-CPose2D CMultiMetricMapPDF_getLastPose(CMultiMetricMapPDF& self, size_t i)
+CPose2D CMultiMetricMapPDF_getLastPose(CMultiMetricMapPDF& me, size_t i)
 {
 	bool is_valid;
-	mrpt::math::TPose3D last_pose = self.getLastPose(i, is_valid);
+	mrpt::math::TPose3D last_pose = me.getLastPose(i, is_valid);
 	return CPose2D(CPose3D(last_pose));
 }
 
-list CMultiMetricMapPDF_getPath(CMultiMetricMapPDF& self, size_t i)
+list CMultiMetricMapPDF_getPath(CMultiMetricMapPDF& me, size_t i)
 {
 	std::deque<mrpt::math::TPose3D> path;
 	list ret_val;
-	self.getPath(i, path);
+	me.getPath(i, path);
 	for (const auto& k : path)
 		ret_val.append(k);
 	return ret_val;
 }
 
 CPose3DPDFParticles CMultiMetricMapPDF_getEstimatedPosePDFAtTime(
-	CMultiMetricMapPDF& self, size_t timeStep)
+	CMultiMetricMapPDF& me, size_t timeStep)
 {
 	CPose3DPDFParticles out_estimation;
-	self.getEstimatedPosePDFAtTime(timeStep, out_estimation);
+	me.getEstimatedPosePDFAtTime(timeStep, out_estimation);
 	return out_estimation;
 }
 // end of CMultiMetricMapPDF
 
 // TSetOfMetricMapInitializers
 void TSetOfMetricMapInitializers_push_back(
-	TSetOfMetricMapInitializers& self, TMetricMapInitializer::Ptr& o)
+	TSetOfMetricMapInitializers& me, TMetricMapInitializer::Ptr& o)
 {
-	self.push_back(o);
+	me.push_back(o);
 }
 // end of TSetOfMetricMapInitializers
 
 // CMultiMetricMap
 mrpt::opengl::CSetOfObjects::Ptr CMultiMetricMap_getAs3DObject(
-	CMultiMetricMap& self)
+	CMultiMetricMap& me)
 {
-	mrpt::opengl::CSetOfObjects::Ptr outObj =
-		mrpt::opengl::CSetOfObjects::Create();
-	self.getAs3DObject(outObj);
-	return outObj;
+	return me.getVisualization();
 }
 
 void CMultiMetricMap_setListOfMaps(
-	CMultiMetricMap& self, TSetOfMetricMapInitializers& initializers)
+	CMultiMetricMap& me, TSetOfMetricMapInitializers& initializers)
 {
-	self.setListOfMaps(initializers);
+	me.setListOfMaps(initializers);
 }
 
-CSimplePointsMap::Ptr CMultiMetricMap_getAsSimplePointsMap(
-	CMultiMetricMap& self)
+CSimplePointsMap::Ptr CMultiMetricMap_getAsSimplePointsMap(CMultiMetricMap& me)
 {
 	CSimplePointsMap::Ptr points_map(new CSimplePointsMap);
-	const CSimplePointsMap* points_map_ptr = self.getAsSimplePointsMap();
+	const CSimplePointsMap* points_map_ptr = me.getAsSimplePointsMap();
 	*points_map = *points_map_ptr;
 	return points_map;
 }

--- a/python/src/math_bindings.cpp
+++ b/python/src/math_bindings.cpp
@@ -33,13 +33,13 @@
 
 #define MAKE_FIXED_DOUBLE_MATRIX_GETITEM(rows, cols)                           \
 	double CMatrixDouble##rows##cols##_getitem1(                               \
-		CMatrixDouble##rows##cols& self, size_t i)                             \
+		CMatrixDouble##rows##cols& me, size_t i)                               \
 	{                                                                          \
-		return self(i);                                                        \
+		return me(i);                                                          \
 	}                                                                          \
                                                                                \
 	double CMatrixDouble##rows##cols##_getitem2(                               \
-		CMatrixDouble##rows##cols& self, const tuple& t)                       \
+		CMatrixDouble##rows##cols& me, const tuple& t)                         \
 	{                                                                          \
 		extract<int> eir(t[0]);                                                \
 		extract<int> eic(t[1]);                                                \
@@ -49,7 +49,7 @@
 			int col = eic();                                                   \
 			printf(STRINGIFY(rows cols % i % i), row, col);                    \
 			if ((0 <= row && row < rows) && (0 <= col && col < cols))          \
-				return self(row, col);                                         \
+				return me(row, col);                                           \
 			else                                                               \
 			{                                                                  \
 				printf("IndexError");                                          \
@@ -60,20 +60,20 @@
 		{                                                                      \
 			TypeError("Indeces must be integers!");                            \
 		}                                                                      \
-		return self(0, 0);                                                     \
+		return me(0, 0);                                                       \
 	}
 
 #define MAKE_FIXED_DOUBLE_MATRIX_SETITEM(rows, cols)                           \
 	void CMatrixDouble##rows##cols##_setitem1(                                 \
-		CMatrixDouble##rows##cols& self, size_t i, double value)               \
+		CMatrixDouble##rows##cols& me, size_t i, double value)                 \
 	{                                                                          \
-		self(i) = value;                                                       \
+		me(i) = value;                                                         \
 	}                                                                          \
                                                                                \
 	void CMatrixDouble##rows##cols##_setitem2(                                 \
-		CMatrixDouble##rows##cols& self, const tuple& t, double value)         \
+		CMatrixDouble##rows##cols& me, const tuple& t, double value)           \
 	{                                                                          \
-		self(extract<int>(t[0]), extract<int>(t[1])) = value;                  \
+		me(extract<int>(t[0]), extract<int>(t[1])) = value;                    \
 	}
 
 #define MAKE_FIXED_DOUBLE_MATRIX_GETSET(rows, cols)                            \

--- a/python/src/nav_bindings.cpp
+++ b/python/src/nav_bindings.cpp
@@ -21,14 +21,14 @@ using namespace mrpt::maps;
 
 // PlannerSimple2D
 tuple PlannerSimple2D_computePath(
-	PlannerSimple2D& self, COccupancyGridMap2D& theMap, CPose2D& origin,
+	PlannerSimple2D& me, COccupancyGridMap2D& theMap, CPose2D& origin,
 	CPose2D& target)
 {
 	// function args
 	std::deque<mrpt::math::TPoint2D> path;
 	bool notFound;
 	// invoke function
-	self.computePath(theMap, origin, target, path, notFound);
+	me.computePath(theMap, origin, target, path, notFound);
 	// convert to python compatible
 	boost::python::list ret_val;
 	boost::python::list py_path;

--- a/python/src/obs_bindings.cpp
+++ b/python/src/obs_bindings.cpp
@@ -37,74 +37,74 @@ using namespace mrpt::serialization;
 
 // CActionCollection
 void CActionCollection_insert1(
-	CActionCollection& self, CActionRobotMovement2D& action)
+	CActionCollection& me, CActionRobotMovement2D& action)
 {
-	self.insert(action);
+	me.insert(action);
 }
 
 void CActionCollection_insert2(
-	CActionCollection& self, CActionRobotMovement3D& action)
+	CActionCollection& me, CActionRobotMovement3D& action)
 {
-	self.insert(action);
+	me.insert(action);
 }
 // end of CActionCollection
 
 // CSensoryFrame
-const CPointsMap* CSensoryFrame_getAuxPointsMap(CSensoryFrame& self)
+const CPointsMap* CSensoryFrame_getAuxPointsMap(CSensoryFrame& me)
 {
-	return self.getAuxPointsMap<CPointsMap>();
+	return me.getAuxPointsMap<CPointsMap>();
 }
 
-const CPointsMap* CSensoryFrame_buildAuxPointsMap(CSensoryFrame& self)
+const CPointsMap* CSensoryFrame_buildAuxPointsMap(CSensoryFrame& me)
 {
-	return self.buildAuxPointsMap<CPointsMap>();
+	return me.buildAuxPointsMap<CPointsMap>();
 }
 // end of CSensoryFrame
 
 // CObservation
-long_ CObservation_get_timestamp(CObservation& self)
+long_ CObservation_get_timestamp(CObservation& me)
 {
-	return long_(self.timestamp);
+	return long_(me.timestamp);
 }
 
-void CObservation_set_timestamp(CObservation& self, long_ timestamp)
+void CObservation_set_timestamp(CObservation& me, long_ timestamp)
 {
 	auto t = mrpt::Clock::fromDouble(extract<uint64_t>(timestamp));
-	self.timestamp = t;
+	me.timestamp = t;
 }
 
-CPose3D CObservation_getSensorPose1(CObservation& self)
+CPose3D CObservation_getSensorPose1(CObservation& me)
 {
 	CPose3D sensorPose;
-	self.getSensorPose(sensorPose);
+	me.getSensorPose(sensorPose);
 	return sensorPose;
 }
 
-void CObservation_getSensorPose2(CObservation& self, CPose3D& sensorPose)
+void CObservation_getSensorPose2(CObservation& me, CPose3D& sensorPose)
 {
-	self.getSensorPose(sensorPose);
+	me.getSensorPose(sensorPose);
 }
 
-void CObservation_getSensorPose3(CObservation& self, TPose3D& sensorPose)
+void CObservation_getSensorPose3(CObservation& me, TPose3D& sensorPose)
 {
-	self.getSensorPose(sensorPose);
+	me.getSensorPose(sensorPose);
 }
 
-void CObservation_setSensorPose1(CObservation& self, CPose3D& sensorPose)
+void CObservation_setSensorPose1(CObservation& me, CPose3D& sensorPose)
 {
-	self.setSensorPose(sensorPose);
+	me.setSensorPose(sensorPose);
 }
 
-void CObservation_setSensorPose2(CObservation& self, TPose3D& sensorPose)
+void CObservation_setSensorPose2(CObservation& me, TPose3D& sensorPose)
 {
-	self.setSensorPose(sensorPose);
+	me.setSensorPose(sensorPose);
 }
 // end of CObservation
 
 // CObservationOdometry
 #ifdef ROS_EXTENSIONS
 object CObservationOdometry_to_ROS_RawOdometry_msg(
-	CObservationOdometry& self, str frame_id)
+	CObservationOdometry& me, str frame_id)
 {
 	// import msg
 	dict locals;
@@ -116,38 +116,37 @@ object CObservationOdometry_to_ROS_RawOdometry_msg(
 	// set info
 	raw_odometry_msg.attr("header").attr("frame_id") = frame_id;
 	raw_odometry_msg.attr("header").attr("stamp") =
-		TTimeStamp_to_ROS_Time(long_(self.timestamp));
-	raw_odometry_msg.attr("has_encoders_info") = self.hasEncodersInfo;
-	raw_odometry_msg.attr("has_velocities") = self.hasVelocities;
+		TTimeStamp_to_ROS_Time(long_(me.timestamp));
+	raw_odometry_msg.attr("has_encoders_info") = me.hasEncodersInfo;
+	raw_odometry_msg.attr("has_velocities") = me.hasVelocities;
 	// set data
-	raw_odometry_msg.attr("encoder_left_ticks") = self.encoderLeftTicks;
-	raw_odometry_msg.attr("encoder_right_ticks") = self.encoderRightTicks;
-	raw_odometry_msg.attr("velocity_lin") = self.velocityLocal.vx;
-	raw_odometry_msg.attr("velocity_ang") = self.velocityLocal.omega;
+	raw_odometry_msg.attr("encoder_left_ticks") = me.encoderLeftTicks;
+	raw_odometry_msg.attr("encoder_right_ticks") = me.encoderRightTicks;
+	raw_odometry_msg.attr("velocity_lin") = me.velocityLocal.vx;
+	raw_odometry_msg.attr("velocity_ang") = me.velocityLocal.omega;
 	return raw_odometry_msg;
 }
 
 void CObservationOdometry_from_ROS_RawOdometry_msg(
-	CObservationOdometry& self, object raw_odometry_msg)
+	CObservationOdometry& me, object raw_odometry_msg)
 {
 	// set info
-	self.sensorLabel =
+	me.sensorLabel =
 		extract<std::string>(raw_odometry_msg.attr("header").attr("frame_id"));
 	auto t = mrpt::Clock::fromDouble(extract<uint64_t>(TTimeStamp_from_ROS_Time(
 		raw_odometry_msg.attr("header").attr("stamp"))));
-	self.timestamp = t;
-	self.hasEncodersInfo =
+	me.timestamp = t;
+	me.hasEncodersInfo =
 		extract<bool>(raw_odometry_msg.attr("has_encoders_info"));
-	self.hasVelocities = extract<bool>(raw_odometry_msg.attr("has_velocities"));
+	me.hasVelocities = extract<bool>(raw_odometry_msg.attr("has_velocities"));
 	// set data
-	self.encoderLeftTicks =
+	me.encoderLeftTicks =
 		extract<int>(raw_odometry_msg.attr("encoder_left_ticks"));
-	self.encoderRightTicks =
+	me.encoderRightTicks =
 		extract<int>(raw_odometry_msg.attr("encoder_right_ticks"));
-	self.velocityLocal.vx =
-		extract<float>(raw_odometry_msg.attr("velocity_lin"));
-	self.velocityLocal.vy = 0;
-	self.velocityLocal.omega =
+	me.velocityLocal.vx = extract<float>(raw_odometry_msg.attr("velocity_lin"));
+	me.velocityLocal.vy = 0;
+	me.velocityLocal.omega =
 		extract<float>(raw_odometry_msg.attr("velocity_ang"));
 }
 #endif
@@ -155,7 +154,7 @@ void CObservationOdometry_from_ROS_RawOdometry_msg(
 
 // CObservationRange
 #ifdef ROS_EXTENSIONS
-object CObservationRange_to_ROS_Range_msg(CObservationRange& self, str frame_id)
+object CObservationRange_to_ROS_Range_msg(CObservationRange& me, str frame_id)
 {
 	// import msg
 	dict locals;
@@ -167,32 +166,32 @@ object CObservationRange_to_ROS_Range_msg(CObservationRange& self, str frame_id)
 	// set info
 	range_msg.attr("header").attr("frame_id") = frame_id;
 	range_msg.attr("header").attr("stamp") =
-		TTimeStamp_to_ROS_Time(long_(self.timestamp));
-	range_msg.attr("min_range") = self.minSensorDistance;
-	range_msg.attr("max_range") = self.maxSensorDistance;
-	range_msg.attr("field_of_view") = self.sensorConeApperture;
+		TTimeStamp_to_ROS_Time(long_(me.timestamp));
+	range_msg.attr("min_range") = me.minSensorDistance;
+	range_msg.attr("max_range") = me.maxSensorDistance;
+	range_msg.attr("field_of_view") = me.sensorConeApperture;
 	// set range
-	range_msg.attr("range") = self.sensedData[0].sensedDistance;
+	range_msg.attr("range") = me.sensedData[0].sensedDistance;
 	return range_msg;
 }
 
 void CObservationRange_from_ROS_Range_msg(
-	CObservationRange& self, object range_msg)
+	CObservationRange& me, object range_msg)
 {
 	// set info
-	self.sensorLabel =
+	me.sensorLabel =
 		extract<std::string>(range_msg.attr("header").attr("frame_id"));
 	auto t = mrpt::Clock::fromDouble(extract<uint64_t>(
 		TTimeStamp_from_ROS_Time(range_msg.attr("header").attr("stamp"))));
-	self.timestamp = t;
-	self.minSensorDistance = extract<float>(range_msg.attr("min_range"));
-	self.maxSensorDistance = extract<float>(range_msg.attr("max_range"));
-	self.sensorConeApperture = extract<float>(range_msg.attr("field_of_view"));
+	me.timestamp = t;
+	me.minSensorDistance = extract<float>(range_msg.attr("min_range"));
+	me.maxSensorDistance = extract<float>(range_msg.attr("max_range"));
+	me.sensorConeApperture = extract<float>(range_msg.attr("field_of_view"));
 	// set range
 	CObservationRange::TMeasurement data;
 	data.sensedDistance = extract<float>(range_msg.attr("range"));
-	self.sensedData.clear();
-	self.sensedData.push_back(data);
+	me.sensedData.clear();
+	me.sensedData.push_back(data);
 }
 #endif
 // end of CObservationRange
@@ -200,7 +199,7 @@ void CObservationRange_from_ROS_Range_msg(
 // CObservation2DRangeScan
 #ifdef ROS_EXTENSIONS
 object CObservation2DRangeScan_to_ROS_LaserScan_msg(
-	CObservation2DRangeScan& self, str frame_id)
+	CObservation2DRangeScan& me, str frame_id)
 {
 	// import msg
 	dict locals;
@@ -212,43 +211,43 @@ object CObservation2DRangeScan_to_ROS_LaserScan_msg(
 	// set info
 	scan_msg.attr("header").attr("frame_id") = frame_id;
 	scan_msg.attr("header").attr("stamp") =
-		TTimeStamp_to_ROS_Time(long_(self.timestamp));
+		TTimeStamp_to_ROS_Time(long_(me.timestamp));
 	scan_msg.attr("range_min") = 0.0;
-	scan_msg.attr("range_max") = self.maxRange;
-	scan_msg.attr("angle_min") = -self.aperture / 2.0;
-	scan_msg.attr("angle_max") = self.aperture / 2.0;
-	scan_msg.attr("angle_increment") = self.beamAperture;
+	scan_msg.attr("range_max") = me.maxRange;
+	scan_msg.attr("angle_min") = -me.aperture / 2.0;
+	scan_msg.attr("angle_max") = me.aperture / 2.0;
+	scan_msg.attr("angle_increment") = me.beamAperture;
 	// set ranges (no intensities given in mrpt)
 	list ranges;
-	for (size_t i = 0; i < self.getScanSize(); i++)
-		ranges.append(self.getScanRange(i));
+	for (size_t i = 0; i < me.getScanSize(); i++)
+		ranges.append(me.getScanRange(i));
 	scan_msg.attr("ranges") = ranges;
 	return scan_msg;
 }
 
 void CObservation2DRangeScan_from_ROS_LaserScan_msg(
-	CObservation2DRangeScan& self, object scan_msg, CPose3D pose)
+	CObservation2DRangeScan& me, object scan_msg, CPose3D pose)
 {
 	// set info
-	self.sensorLabel =
+	me.sensorLabel =
 		extract<std::string>(scan_msg.attr("header").attr("frame_id"));
 	auto t = mrpt::Clock::fromDouble(extract<uint64_t>(
 		TTimeStamp_from_ROS_Time(scan_msg.attr("header").attr("stamp"))));
-	self.timestamp = t;
-	self.maxRange = extract<float>(scan_msg.attr("range_max"));
-	self.aperture = extract<float>(scan_msg.attr("angle_max")) -
+	me.timestamp = t;
+	me.maxRange = extract<float>(scan_msg.attr("range_max"));
+	me.aperture = extract<float>(scan_msg.attr("angle_max")) -
 		extract<float>(scan_msg.attr("angle_min"));
-	self.beamAperture = extract<float>(scan_msg.attr("angle_increment"));
-	self.sensorPose = pose;
+	me.beamAperture = extract<float>(scan_msg.attr("angle_increment"));
+	me.sensorPose = pose;
 	// set ranges
 	tuple ranges = extract<tuple>(scan_msg.attr("ranges"));
 	const size_t N = len(ranges);
-	self.resizeScan(N);
+	me.resizeScan(N);
 	for (int i = 0; i < len(ranges); ++i)
 	{
 		float range = extract<float>(ranges[i]);
-		self.setScanRange(i, range);
-		self.setScanRangeValidity(i, (range < self.maxRange - 0.01));
+		me.setScanRange(i, range);
+		me.setScanRangeValidity(i, (range < me.maxRange - 0.01));
 	}
 }
 #endif

--- a/python/src/opengl_bindings.cpp
+++ b/python/src/opengl_bindings.cpp
@@ -32,30 +32,30 @@ using namespace mrpt::serialization;
 using namespace mrpt::io;
 
 // CRenderizable
-void CRenderizable_setPose(CRenderizable& self, mrpt::poses::CPose3D pose)
+void CRenderizable_setPose(CRenderizable& me, mrpt::poses::CPose3D pose)
 {
-	self.setPose(pose);
+	me.setPose(pose);
 }
 
-void CRenderizable_setLocation1(CRenderizable& self, mrpt::poses::CPoint3D pose)
+void CRenderizable_setLocation1(CRenderizable& me, mrpt::poses::CPoint3D pose)
 {
-	self.setLocation(pose.asTPoint());
+	me.setLocation(pose.asTPoint());
 }
 
-void CRenderizable_setLocation2(CRenderizable& self, float x, float y, float z)
+void CRenderizable_setLocation2(CRenderizable& me, float x, float y, float z)
 {
-	self.setLocation(x, y, z);
+	me.setLocation(x, y, z);
 }
 
-void CRenderizable_setColor1(CRenderizable& self, mrpt::img::TColorf& c)
+void CRenderizable_setColor1(CRenderizable& me, mrpt::img::TColorf& c)
 {
-	self.setColor(c);
+	me.setColor(c);
 }
 
 void CRenderizable_setColor2(
-	CRenderizable& self, double r, double g, double b, double alpha = 1.0)
+	CRenderizable& me, double r, double g, double b, double alpha = 1.0)
 {
-	self.setColor(r, g, b, alpha);
+	me.setColor(r, g, b, alpha);
 }
 
 BOOST_PYTHON_FUNCTION_OVERLOADS(
@@ -64,10 +64,10 @@ BOOST_PYTHON_FUNCTION_OVERLOADS(
 
 // COpenGLScene
 void COpenGLScene_insert(
-	COpenGLScene& self, const CRenderizable::Ptr& newObject,
+	COpenGLScene& me, const CRenderizable::Ptr& newObject,
 	const std::string& viewportName = std::string("main"))
 {
-	self.insert(newObject, viewportName);
+	me.insert(newObject, viewportName);
 }
 
 BOOST_PYTHON_FUNCTION_OVERLOADS(
@@ -78,10 +78,9 @@ BOOST_PYTHON_MEMBER_FUNCTION_OVERLOADS(
 
 // CSetOfLines
 void CSetOfLines_appendLine(
-	CSetOfLines& self, float x0, float y0, float z0, float x1, float y1,
-	float z1)
+	CSetOfLines& me, float x0, float y0, float z0, float x1, float y1, float z1)
 {
-	self.appendLine(x0, y0, z0, x1, y1, z1);
+	me.appendLine(x0, y0, z0, x1, y1, z1);
 }
 
 CSetOfLines::Ptr CSetOfLines_Create()
@@ -95,14 +94,14 @@ CEllipsoid3D::Ptr CEllipsoid3D_Create()
 {
 	return std::make_shared<CEllipsoid3D>();
 }
-void CEllipsoid3D_setFromPosePDF(CEllipsoid3D& self, CPose3DPDF& posePDF)
+void CEllipsoid3D_setFromPosePDF(CEllipsoid3D& me, CPose3DPDF& posePDF)
 {
 	CPose3D meanPose;
 	CMatrixDouble66 COV;
 	posePDF.getCovarianceAndMean(COV, meanPose);
 	CMatrixDouble33 COV3 = COV.extractMatrix<3, 3>(0, 0);
-	self.setLocation(meanPose.x(), meanPose.y(), meanPose.z() + 0.001);
-	self.setCovMatrix(COV3);
+	me.setLocation(meanPose.x(), meanPose.y(), meanPose.z() + 0.001);
+	me.setCovMatrix(COV3);
 }
 // end of CEllipsoid3D
 
@@ -111,14 +110,14 @@ CEllipsoid2D::Ptr CEllipsoid2D_Create()
 {
 	return std::make_shared<CEllipsoid2D>();
 }
-void CEllipsoid2D_setFromPosePDF(CEllipsoid2D& self, CPose3DPDF& posePDF)
+void CEllipsoid2D_setFromPosePDF(CEllipsoid2D& me, CPose3DPDF& posePDF)
 {
 	CPose3D meanPose;
 	CMatrixDouble66 COV;
 	posePDF.getCovarianceAndMean(COV, meanPose);
 	CMatrixDouble22 COV2 = COV.extractMatrix<2, 2>(0, 0);
-	self.setLocation(meanPose.x(), meanPose.y(), meanPose.z() + 0.001);
-	self.setCovMatrix(COV2);
+	me.setLocation(meanPose.x(), meanPose.y(), meanPose.z() + 0.001);
+	me.setCovMatrix(COV2);
 }
 // end of CEllipsoid2D
 

--- a/python/src/poses_bindings.cpp
+++ b/python/src/poses_bindings.cpp
@@ -29,8 +29,8 @@ using namespace mrpt::poses;
 
 // prototypes
 #ifdef ROS_EXTENSIONS
-object CPose3D_to_ROS_Pose_msg(CPose3D& self);
-void CPose3D_from_ROS_Pose_msg(CPose3D& self, object pose_msg);
+object CPose3D_to_ROS_Pose_msg(CPose3D& me);
+void CPose3D_from_ROS_Pose_msg(CPose3D& me, object pose_msg);
 #endif
 // end of prototypes
 
@@ -45,17 +45,17 @@ void (CPose2D::*CPose2D_set_phi)(double) = &CPose2D::phi;
 MAKE_AS_STR(CPose2D)
 
 #ifdef ROS_EXTENSIONS
-object CPose2D_to_ROS_Pose_msg(CPose2D& self)
+object CPose2D_to_ROS_Pose_msg(CPose2D& me)
 {
-	CPose3D pose(self);
+	CPose3D pose(me);
 	return CPose3D_to_ROS_Pose_msg(pose);
 }
 
-void CPose2D_from_ROS_Pose_msg(CPose2D& self, object pose_msg)
+void CPose2D_from_ROS_Pose_msg(CPose2D& me, object pose_msg)
 {
 	CPose3D pose;
 	CPose3D_from_ROS_Pose_msg(pose, pose_msg);
-	self = CPose2D(pose);
+	me = CPose2D(pose);
 }
 #endif
 // end of CPose2D
@@ -68,11 +68,11 @@ void (CPose3D::*CPose3D_set_y)(double) = &CPose3D::y;
 double& (CPose3D::*CPose3D_get_z)() = &CPose3D::z;
 void (CPose3D::*CPose3D_set_z)(double) = &CPose3D::z;
 
-tuple CPose3D_getYawPitchRoll(CPose3D& self)
+tuple CPose3D_getYawPitchRoll(CPose3D& me)
 {
 	list ret_val;
 	double yaw, pitch, roll;
-	self.getYawPitchRoll(yaw, pitch, roll);
+	me.getYawPitchRoll(yaw, pitch, roll);
 	ret_val.append(yaw);
 	ret_val.append(pitch);
 	ret_val.append(roll);
@@ -83,9 +83,9 @@ MAKE_AS_STR(CPose3D)
 MAKE_GETITEM(CPose3D, double)
 
 #ifdef ROS_EXTENSIONS
-object CPose3D_to_ROS_Pose_msg(CPose3D& self)
+object CPose3D_to_ROS_Pose_msg(CPose3D& me)
 {
-	CPose3DQuat pose_quat(self);
+	CPose3DQuat pose_quat(me);
 	// import msg
 	dict locals;
 	exec(
@@ -103,7 +103,7 @@ object CPose3D_to_ROS_Pose_msg(CPose3D& self)
 	return pose_msg;
 }
 
-void CPose3D_from_ROS_Pose_msg(CPose3D& self, object pose_msg)
+void CPose3D_from_ROS_Pose_msg(CPose3D& me, object pose_msg)
 {
 	CPose3DQuat pose_quat;
 	pose_quat[0] = extract<double>(pose_msg.attr("position").attr("x"));
@@ -113,7 +113,7 @@ void CPose3D_from_ROS_Pose_msg(CPose3D& self, object pose_msg)
 	pose_quat[5] = extract<double>(pose_msg.attr("orientation").attr("y"));
 	pose_quat[6] = extract<double>(pose_msg.attr("orientation").attr("z"));
 	pose_quat[3] = extract<double>(pose_msg.attr("orientation").attr("w"));
-	self = CPose3D(pose_quat);
+	me = CPose3D(pose_quat);
 }
 #endif
 // end of CPose3D
@@ -254,51 +254,50 @@ void CPose3D_from_ROS_Pose_msg(CPose3D& self, object pose_msg)
 // };
 
 // CPosePDFGaussian
-list CPosePDFGaussian_get_cov(CPosePDFGaussian& self)
+list CPosePDFGaussian_get_cov(CPosePDFGaussian& me)
 {
 	list cov;
 	for (int32_t i = 0; i < 9; ++i)
 	{
-		cov.append(self.cov(i));
+		cov.append(me.cov(i));
 	}
 	return cov;
 }
 
-void CPosePDFGaussian_set_cov(CPosePDFGaussian& self, list cov)
+void CPosePDFGaussian_set_cov(CPosePDFGaussian& me, list cov)
 {
 	for (int32_t i = 0; i < 9; ++i)
 	{
-		self.cov(i) = extract<double>(cov[i]);
+		me.cov(i) = extract<double>(cov[i]);
 	}
 }
 // end of CPosePDFGaussian
 
 // CPose3DPDFGaussian
-list CPose3DPDFGaussian_get_cov(CPose3DPDFGaussian& self)
+list CPose3DPDFGaussian_get_cov(CPose3DPDFGaussian& me)
 {
 	list cov;
 	for (int32_t i = 0; i < 36; ++i)
 	{
-		cov.append(self.cov(i));
+		cov.append(me.cov(i));
 	}
 	return cov;
 }
 
-void CPose3DPDFGaussian_set_cov(CPose3DPDFGaussian& self, list cov)
+void CPose3DPDFGaussian_set_cov(CPose3DPDFGaussian& me, list cov)
 {
 	for (int32_t i = 0; i < 36; ++i)
 	{
-		self.cov(i) = extract<double>(cov[i]);
+		me.cov(i) = extract<double>(cov[i]);
 	}
 }
 
 MAKE_AS_STR(CPose3DPDFGaussian)
 
 #ifdef ROS_EXTENSIONS
-object CPose3DPDFGaussian_to_ROS_PoseWithCovariance_msg(
-	CPose3DPDFGaussian& self)
+object CPose3DPDFGaussian_to_ROS_PoseWithCovariance_msg(CPose3DPDFGaussian& me)
 {
-	CPose3DQuat pose_quat(self.mean);
+	CPose3DQuat pose_quat(me.mean);
 	// import msg
 	dict locals;
 	exec(
@@ -306,7 +305,7 @@ object CPose3DPDFGaussian_to_ROS_PoseWithCovariance_msg(
 		"pose_msg = PoseWithCovariance()\n",
 		object(), locals);
 	object pose_msg = locals["pose_msg"];
-	pose_msg.attr("pose") = CPose3D_to_ROS_Pose_msg(self.mean);
+	pose_msg.attr("pose") = CPose3D_to_ROS_Pose_msg(me.mean);
 
 	// Read REP103: http://ros.org/reps/rep-0103.html#covariance-representation
 	// # Row-major representation of the 6x6 covariance matrix
@@ -323,7 +322,7 @@ object CPose3DPDFGaussian_to_ROS_PoseWithCovariance_msg(
 	const unsigned int indxs_map[6] = {0, 1, 2,
 									   5, 4, 3};  // X,Y,Z,YAW,PITCH,ROLL
 
-	list mrptcov = CPose3DPDFGaussian_get_cov(self);
+	list mrptcov = CPose3DPDFGaussian_get_cov(me);
 	pose_msg.attr("covariance") = list(mrptcov);
 
 	for (int i = 0; i < 6; i++)
@@ -339,10 +338,10 @@ object CPose3DPDFGaussian_to_ROS_PoseWithCovariance_msg(
 }
 
 void CPose3DPDFGaussian_from_ROS_PoseWithCovariance_msg(
-	CPose3DPDFGaussian& self, object pose_msg)
+	CPose3DPDFGaussian& me, object pose_msg)
 {
-	self = CPose3DPDFGaussian();
-	CPose3D_from_ROS_Pose_msg(self.mean, pose_msg.attr("pose"));
+	me = CPose3DPDFGaussian();
+	CPose3D_from_ROS_Pose_msg(me.mean, pose_msg.attr("pose"));
 
 	list roscov = list(pose_msg.attr("covariance"));
 
@@ -354,7 +353,7 @@ void CPose3DPDFGaussian_from_ROS_PoseWithCovariance_msg(
 	{
 		for (int j = 0; j < 6; j++)
 		{
-			self.cov(i, j) =
+			me.cov(i, j) =
 				extract<double>(roscov[indxs_map[i] * 6 + indxs_map[j]]);
 		}
 	}

--- a/python/src/slam_bindings.cpp
+++ b/python/src/slam_bindings.cpp
@@ -46,14 +46,14 @@ using namespace mrpt::io;
 
 // CICP
 tuple CICP_AlignPDF1(
-	CICP& self, COccupancyGridMap2D& m1, CSimplePointsMap& m2,
+	CICP& me, COccupancyGridMap2D& m1, CSimplePointsMap& m2,
 	CPosePDFGaussian& initialEstimationPDF)
 {
 	CPosePDFGaussian posePDF;
 	CICP::TReturnInfo info;
 
 	CPosePDF::Ptr posePDFPtr =
-		self.AlignPDF(&m1, &m2, initialEstimationPDF, info);
+		me.AlignPDF(&m1, &m2, initialEstimationPDF, info);
 	posePDF.copyFrom(*posePDFPtr);
 
 	boost::python::list ret_val;
@@ -63,14 +63,14 @@ tuple CICP_AlignPDF1(
 }
 
 tuple CICP_AlignPDF2(
-	CICP& self, CSimplePointsMap& m1, CSimplePointsMap& m2,
+	CICP& me, CSimplePointsMap& m1, CSimplePointsMap& m2,
 	CPosePDFGaussian& initialEstimationPDF)
 {
 	CPosePDFGaussian posePDF;
 	CICP::TReturnInfo info;
 
 	CPosePDF::Ptr posePDFPtr =
-		self.AlignPDF(&m1, &m2, initialEstimationPDF, info);
+		me.AlignPDF(&m1, &m2, initialEstimationPDF, info);
 	posePDF.copyFrom(*posePDFPtr);
 
 	boost::python::list ret_val;
@@ -112,40 +112,40 @@ struct CMetricMapBuilderWrap : CMetricMapBuilder, wrapper<CMetricMapBuilder>
 	}
 };
 
-CSimpleMap CMetricMapBuilder_getCurrentlyBuiltMap(CMetricMapBuilder& self)
+CSimpleMap CMetricMapBuilder_getCurrentlyBuiltMap(CMetricMapBuilder& me)
 {
 	CSimpleMap out_map;
-	self.getCurrentlyBuiltMap(out_map);
+	me.getCurrentlyBuiltMap(out_map);
 	return out_map;
 }
 
-CPose2D CMetricMapBuilder_getCurrentPose2D(CMetricMapBuilder& self)
+CPose2D CMetricMapBuilder_getCurrentPose2D(CMetricMapBuilder& me)
 {
-	return CPose2D(self.getCurrentPoseEstimation()->getMeanVal());
+	return CPose2D(me.getCurrentPoseEstimation()->getMeanVal());
 }
 
 void CMetricMapBuilderICP_initialize(
-	CMetricMapBuilderICP& self, CSimpleMap& initialMap, CPosePDFGaussian& x0)
+	CMetricMapBuilderICP& me, CSimpleMap& initialMap, CPosePDFGaussian& x0)
 {
-	self.initialize(initialMap, &x0);
+	me.initialize(initialMap, &x0);
 }
 
 void CMetricMapBuilderRBPF_initialize(
-	CMetricMapBuilderRBPF& self, CSimpleMap& initialMap, CPosePDFParticles& x0)
+	CMetricMapBuilderRBPF& me, CSimpleMap& initialMap, CPosePDFParticles& x0)
 {
-	self.initialize(initialMap, &x0);
+	me.initialize(initialMap, &x0);
 }
 
 CPose3DPDFParticles::Ptr CMetricMapBuilderRBPF_getCurrentPoseEstimation(
-	CMetricMapBuilderRBPF& self)
+	CMetricMapBuilderRBPF& me)
 {
 	return std::dynamic_pointer_cast<CPose3DPDFParticles>(
-		self.getCurrentPoseEstimation());
+		me.getCurrentPoseEstimation());
 }
 // end of CMetricMapBuilder
 
 // CRangeBearingKFSLAM2D
-tuple CRangeBearingKFSLAM2D_getCurrentState(CRangeBearingKFSLAM2D& self)
+tuple CRangeBearingKFSLAM2D_getCurrentState(CRangeBearingKFSLAM2D& me)
 {
 	list ret_val;
 
@@ -155,7 +155,7 @@ tuple CRangeBearingKFSLAM2D_getCurrentState(CRangeBearingKFSLAM2D& self)
 	mrpt::math::CVectorDouble out_fullState;
 	mrpt::math::CMatrixDouble out_fullCovariance;
 
-	self.getCurrentState(
+	me.getCurrentState(
 		out_robotPose, out_landmarksPositions, out_landmarkIDs, out_fullState,
 		out_fullCovariance);
 
@@ -169,19 +169,19 @@ tuple CRangeBearingKFSLAM2D_getCurrentState(CRangeBearingKFSLAM2D& self)
 }
 
 CPosePDFGaussian CRangeBearingKFSLAM2D_getCurrentRobotPose(
-	CRangeBearingKFSLAM2D& self)
+	CRangeBearingKFSLAM2D& me)
 {
 	CPosePDFGaussian out_robotPose;
-	self.getCurrentRobotPose(out_robotPose);
+	me.getCurrentRobotPose(out_robotPose);
 	return out_robotPose;
 }
 
 mrpt::opengl::CSetOfObjects::Ptr CRangeBearingKFSLAM2D_getAs3DObject(
-	CRangeBearingKFSLAM2D& self)
+	CRangeBearingKFSLAM2D& me)
 {
 	mrpt::opengl::CSetOfObjects::Ptr outObj =
 		mrpt::opengl::CSetOfObjects::Create();
-	self.getAs3DObject(outObj);
+	me.getAs3DObject(outObj);
 	return outObj;
 }
 
@@ -192,106 +192,105 @@ BOOST_PYTHON_MEMBER_FUNCTION_OVERLOADS(
 
 // CMonteCarloLocalization2D
 void CMonteCarloLocalization2D_prediction_and_update_pfStandardProposal(
-	CMonteCarloLocalization2D& self,
+	CMonteCarloLocalization2D& me,
 	const mrpt::obs::CActionCollection::Ptr action,
 	const mrpt::obs::CSensoryFrame::Ptr observation,
 	const CParticleFilter::TParticleFilterOptions& PF_options)
 {
-	self.prediction_and_update_pfStandardProposal(
+	me.prediction_and_update_pfStandardProposal(
 		action.get(), observation.get(), PF_options);
 }
 
 void CMonteCarloLocalization2D_prediction_and_update_pfAuxiliaryPFStandard(
-	CMonteCarloLocalization2D& self,
+	CMonteCarloLocalization2D& me,
 	const mrpt::obs::CActionCollection::Ptr action,
 	const mrpt::obs::CSensoryFrame::Ptr observation,
 	const CParticleFilter::TParticleFilterOptions& PF_options)
 {
-	self.prediction_and_update_pfAuxiliaryPFStandard(
+	me.prediction_and_update_pfAuxiliaryPFStandard(
 		action.get(), observation.get(), PF_options);
 }
 
 void CMonteCarloLocalization2D_prediction_and_update_pfAuxiliaryPFOptimal(
-	CMonteCarloLocalization2D& self,
+	CMonteCarloLocalization2D& me,
 	const mrpt::obs::CActionCollection::Ptr action,
 	const mrpt::obs::CSensoryFrame::Ptr observation,
 	const CParticleFilter::TParticleFilterOptions& PF_options)
 {
-	self.prediction_and_update_pfAuxiliaryPFOptimal(
+	me.prediction_and_update_pfAuxiliaryPFOptimal(
 		action.get(), observation.get(), PF_options);
 }
 
 void CMonteCarloLocalization2D_prediction_and_update(
-	CMonteCarloLocalization2D& self,
+	CMonteCarloLocalization2D& me,
 	const mrpt::obs::CActionCollection::Ptr action,
 	const mrpt::obs::CSensoryFrame::Ptr observation,
 	const CParticleFilter::TParticleFilterOptions& PF_options)
 {
-	self.prediction_and_update(action.get(), observation.get(), PF_options);
+	me.prediction_and_update(action.get(), observation.get(), PF_options);
 }
 
-CPose2D CMonteCarloLocalization2D_getMean(CMonteCarloLocalization2D& self)
+CPose2D CMonteCarloLocalization2D_getMean(CMonteCarloLocalization2D& me)
 {
 	CPose2D mean;
-	self.getMean(mean);
+	me.getMean(mean);
 	return mean;
 }
 
 tuple CMonteCarloLocalization2D_getCovarianceAndMean(
-	CMonteCarloLocalization2D& self)
+	CMonteCarloLocalization2D& me)
 {
 	list ret_val;
-	const auto [cov, mean_point] = self.getCovarianceAndMean();
+	const auto [cov, mean_point] = me.getCovarianceAndMean();
 	ret_val.append(cov);
 	ret_val.append(mean_point);
 	return tuple(ret_val);
 }
 
 CPose2D CMonteCarloLocalization2D_drawSingleSample(
-	CMonteCarloLocalization2D& self)
+	CMonteCarloLocalization2D& me)
 {
 	CPose2D outPart;
-	self.drawSingleSample(outPart);
+	me.drawSingleSample(outPart);
 	return outPart;
 }
 
 CPosePDFParticles CMonteCarloLocalization2D_inverse(
-	CMonteCarloLocalization2D& self)
+	CMonteCarloLocalization2D& me)
 {
 	CPosePDFParticles o;
-	self.inverse(o);
+	me.inverse(o);
 	return o;
 }
 
 void CMonteCarloLocalization2D_writeParticlesToStream(
-	CMonteCarloLocalization2D& self, CStream& out)
+	CMonteCarloLocalization2D& me, CStream& out)
 {
 	auto arch = mrpt::serialization::archiveFrom(out);
-	self.writeParticlesToStream(arch);
+	me.writeParticlesToStream(arch);
 }
 
 void CMonteCarloLocalization2D_readParticlesFromStream(
-	CMonteCarloLocalization2D& self, CStream& in)
+	CMonteCarloLocalization2D& me, CStream& in)
 {
 	auto arch = mrpt::serialization::archiveFrom(in);
-	self.readParticlesFromStream(arch);
+	me.readParticlesFromStream(arch);
 }
 
 mrpt::opengl::CSetOfObjects::Ptr CMonteCarloLocalization2D_getAs3DObject(
-	CMonteCarloLocalization2D& self)
+	CMonteCarloLocalization2D& me)
 {
 	mrpt::opengl::CSetOfObjects::Ptr outObj =
 		mrpt::opengl::CSetOfObjects::Create();
-	self.getAs3DObject(outObj);
+	me.getAs3DObject(outObj);
 	return outObj;
 }
 
-tuple CMonteCarloLocalization2D_normalizeWeights(
-	CMonteCarloLocalization2D& self)
+tuple CMonteCarloLocalization2D_normalizeWeights(CMonteCarloLocalization2D& me)
 {
 	list ret_val;
 	double out_max_log_w;
-	double norm = self.normalizeWeights(&out_max_log_w);
+	double norm = me.normalizeWeights(&out_max_log_w);
 	ret_val.append(norm);
 	ret_val.append(out_max_log_w);
 	return tuple(ret_val);
@@ -312,106 +311,105 @@ BOOST_PYTHON_MEMBER_FUNCTION_OVERLOADS(
 
 // CMonteCarloLocalization3D
 void CMonteCarloLocalization3D_prediction_and_update_pfStandardProposal(
-	CMonteCarloLocalization3D& self,
+	CMonteCarloLocalization3D& me,
 	const mrpt::obs::CActionCollection::Ptr action,
 	const mrpt::obs::CSensoryFrame::Ptr observation,
 	const CParticleFilter::TParticleFilterOptions& PF_options)
 {
-	self.prediction_and_update_pfStandardProposal(
+	me.prediction_and_update_pfStandardProposal(
 		action.get(), observation.get(), PF_options);
 }
 
 void CMonteCarloLocalization3D_prediction_and_update_pfAuxiliaryPFStandard(
-	CMonteCarloLocalization3D& self,
+	CMonteCarloLocalization3D& me,
 	const mrpt::obs::CActionCollection::Ptr action,
 	const mrpt::obs::CSensoryFrame::Ptr observation,
 	const CParticleFilter::TParticleFilterOptions& PF_options)
 {
-	self.prediction_and_update_pfAuxiliaryPFStandard(
+	me.prediction_and_update_pfAuxiliaryPFStandard(
 		action.get(), observation.get(), PF_options);
 }
 
 void CMonteCarloLocalization3D_prediction_and_update_pfAuxiliaryPFOptimal(
-	CMonteCarloLocalization3D& self,
+	CMonteCarloLocalization3D& me,
 	const mrpt::obs::CActionCollection::Ptr action,
 	const mrpt::obs::CSensoryFrame::Ptr observation,
 	const CParticleFilter::TParticleFilterOptions& PF_options)
 {
-	self.prediction_and_update_pfAuxiliaryPFOptimal(
+	me.prediction_and_update_pfAuxiliaryPFOptimal(
 		action.get(), observation.get(), PF_options);
 }
 
 void CMonteCarloLocalization3D_prediction_and_update(
-	CMonteCarloLocalization3D& self,
+	CMonteCarloLocalization3D& me,
 	const mrpt::obs::CActionCollection::Ptr action,
 	const mrpt::obs::CSensoryFrame::Ptr observation,
 	const CParticleFilter::TParticleFilterOptions& PF_options)
 {
-	self.prediction_and_update(action.get(), observation.get(), PF_options);
+	me.prediction_and_update(action.get(), observation.get(), PF_options);
 }
 
-CPose3D CMonteCarloLocalization3D_getMean(CMonteCarloLocalization3D& self)
+CPose3D CMonteCarloLocalization3D_getMean(CMonteCarloLocalization3D& me)
 {
 	CPose3D mean;
-	self.getMean(mean);
+	me.getMean(mean);
 	return mean;
 }
 
 tuple CMonteCarloLocalization3D_getCovarianceAndMean(
-	CMonteCarloLocalization3D& self)
+	CMonteCarloLocalization3D& me)
 {
 	list ret_val;
-	const auto [cov, mean_point] = self.getCovarianceAndMean();
+	const auto [cov, mean_point] = me.getCovarianceAndMean();
 	ret_val.append(cov);
 	ret_val.append(mean_point);
 	return tuple(ret_val);
 }
 
 CPose3D CMonteCarloLocalization3D_drawSingleSample(
-	CMonteCarloLocalization3D& self)
+	CMonteCarloLocalization3D& me)
 {
 	CPose3D outPart;
-	self.drawSingleSample(outPart);
+	me.drawSingleSample(outPart);
 	return outPart;
 }
 
 CPose3DPDFParticles CMonteCarloLocalization3D_inverse(
-	CMonteCarloLocalization3D& self)
+	CMonteCarloLocalization3D& me)
 {
 	CPose3DPDFParticles o;
-	self.inverse(o);
+	me.inverse(o);
 	return o;
 }
 
 void CMonteCarloLocalization3D_writeParticlesToStream(
-	CMonteCarloLocalization3D& self, CStream& out)
+	CMonteCarloLocalization3D& me, CStream& out)
 {
 	auto arch = mrpt::serialization::archiveFrom(out);
-	self.writeParticlesToStream(arch);
+	me.writeParticlesToStream(arch);
 }
 
 void CMonteCarloLocalization3D_readParticlesFromStream(
-	CMonteCarloLocalization3D& self, CStream& in)
+	CMonteCarloLocalization3D& me, CStream& in)
 {
 	auto arch = mrpt::serialization::archiveFrom(in);
-	self.readParticlesFromStream(arch);
+	me.readParticlesFromStream(arch);
 }
 
 mrpt::opengl::CSetOfObjects::Ptr CMonteCarloLocalization3D_getAs3DObject(
-	CMonteCarloLocalization3D& self)
+	CMonteCarloLocalization3D& me)
 {
 	mrpt::opengl::CSetOfObjects::Ptr outObj =
 		mrpt::opengl::CSetOfObjects::Create();
-	self.getAs3DObject(outObj);
+	me.getAs3DObject(outObj);
 	return outObj;
 }
 
-tuple CMonteCarloLocalization3D_normalizeWeights(
-	CMonteCarloLocalization3D& self)
+tuple CMonteCarloLocalization3D_normalizeWeights(CMonteCarloLocalization3D& me)
 {
 	list ret_val;
 	double out_max_log_w;
-	double norm = self.normalizeWeights(&out_max_log_w);
+	double norm = me.normalizeWeights(&out_max_log_w);
 	ret_val.append(norm);
 	ret_val.append(out_max_log_w);
 	return tuple(ret_val);

--- a/python/src/utils_bindings.cpp
+++ b/python/src/utils_bindings.cpp
@@ -29,14 +29,14 @@ using namespace mrpt::serialization;
 using namespace mrpt::io;
 
 // CStream
-void CStream_ReadObject(CArchive& self, CSerializable::Ptr& obj)
+void CStream_ReadObject(CArchive& me, CSerializable::Ptr& obj)
 {
-	self.ReadObject(obj.get());
+	me.ReadObject(obj.get());
 }
 
-void CStream_WriteObject(CArchive& self, CSerializable::Ptr& obj)
+void CStream_WriteObject(CArchive& me, CSerializable::Ptr& obj)
 {
-	self.WriteObject(obj.get());
+	me.WriteObject(obj.get());
 }
 // end of CStream
 
@@ -65,9 +65,9 @@ double mrpt_utils_RAD2DEG(double rad) { return mrpt::RAD2DEG(rad); }
 // end of Utils
 
 // smart pointer contents
-CObject& CObjectPtr_get_ctx(CObject::Ptr& self) { return *self; }
-void CObjectPtr_set_ctx(CObject::Ptr& self, const CObject& ctx) { *self = ctx; }
-CObject* CObjectPtr_pointer(CObject::Ptr& self) { return self.get(); }
+CObject& CObjectPtr_get_ctx(CObject::Ptr& me) { return *me; }
+void CObjectPtr_set_ctx(CObject::Ptr& me, const CObject& ctx) { *me = ctx; }
+CObject* CObjectPtr_pointer(CObject::Ptr& me) { return me.get(); }
 MAKE_PTR_CTX(CSerializable)
 
 // exporter

